src/main/java/com/es/phoneshop/web/ProductListPageServlet.java

```diff
@@ -22,10 +28,32 @@ public void init(ServletConfig config) throws ServletException {
 
     @Override
     protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
-        request.setAttribute("products", productService.findProducts());
+        String description = request.getParameter("query");
+        String field = request.getParameter("field");
+        String order = request.getParameter("order");
+        String productToShowPriceHistoryId = request.getParameter("productToShowPriceHistoryId");
+        if (!StringUtils.isBlank(productToShowPriceHistoryId)) {
+            request.setAttribute("productToShowPriceHistory", getProductWithPriceHistoryById(productToShowPriceHistoryId));
+        }
+        request.setAttribute("products", receiveProducts(description, field, order));
         request.getRequestDispatcher("/WEB-INF/pages/productList.jsp").forward(request, response);
     }
 
+    private Product getProductWithPriceHistoryById(String productToShowPriceHistoryId) {
```

Comment:
lets save this to another page maybe, not to plp? To make it more user-friendly and clear for customer

To pdp or even to separate page